tests/test_mock_controller.py

```diff
@@ -2,19 +2,31 @@
 import asynctest
 import logging
 import math
+from unittest.mock import MagicMock
+
 import yaml
 
 from lsst.ts import Dome
+from lsst.ts import salobj
 
 logging.basicConfig(
-    format="%(asctime)s:%(levelname)s:%(name)s:%(message)s", level=logging.INFO
+    format="%(asctime)s:%(levelname)s:%(name)s:%(message)s", level=logging.DEBUG
 )
 
 _NUM_LOUVERS = 34
 _NUM_MON_SENSORS = 16
 _NUM_THERMO_SENSORS = 16
 
 
+# MagicMock doesn't support async coroutines and AsyncMock is not available in Python 3.7 but only in 3.8
+# This is a way to provide a future that can be returned by a mock method and that can awaited in the
+# calling code.
+def async_return(result):
+    f = asyncio.Future()
+    f.set_result(result)
+    return f
```

Comment:
Note that we use the `asynctest` package which has its own asyncio-friendly MagicMock class. It'll be interesting to see if we keep using it for the long run. I was disappointed when I read about the asyncio support in Python 3.8 unittest -- asynctest seems significantly easier to use. But we'll have to move away if it is abandoned.

Reply:
It turns out that MagicMock in asynctest also cannot be used to mock methods that need to be awaited. But using CoroutineMock, as suggested by the docs of asynctest, works. Thanks for the tip!